### PR TITLE
utils: gnd identifier support addtion

### PR DESCRIPTION
**Pull request by bouzlibop**
* Adds support for gnd identifier. (addresses zenodo/zenodo#217)

Signed-off-by: Adrian Pawel Baran <adrian.pawel.baran@cern.ch>

### CODE REVIEW

**Comment by bouzlibop** [comment]
cc @lnielsen

**Comment by jirikuncar** [comment]
@lnielsen can we apply it to all `maint-2.x` branches?

File: invenio/utils/persistentid.py [hunk 1]
```diff
@@ -297,6 +303,18 @@ def is_pmcid(val):
     return pmcid_regexp.match(val)
 
 
+def is_gnd(val):
+    """Test if argument is a GND Identifier."""
+    if val.startswith('http://d-nb.info/gnd/'):
```
Comment (lnielsen) [inline invenio/utils/persistentid.py]: Can you assign this string to a module level variable (similar to the regexs) and use that variable instead (i.e. to ensure any change will be propagated correctly).

File: invenio/utils/persistentid.py [hunk 2]
```diff
@@ -316,6 +334,7 @@ def is_pmcid(val):
     ('ean8', is_ean8),
     ('istc', is_istc),
     ('pmid', is_pmid),
+    ('gnd', is_gnd),
```
Comment (lnielsen) [inline invenio/utils/persistentid.py]: Hmm, probably we should take out ``pmid`` as it matches integers.

File: invenio/utils/persistentid.py [hunk 3]
```diff
@@ -371,6 +390,14 @@ def normalize_orcid(val):
     return "-".join([val[0:4], val[4:8], val[8:12], val[12:16]])
 
 
+def normalize_gnd(val):
+    """Normalize a GND identifier."""
+    if val.startswith("http://d-nb.info/gnd/"):
```
Comment (lnielsen) [inline invenio/utils/persistentid.py]: Here you can assume you have a valid GND identifier. Current a valid GND can only have one form, so I'm about to think we don't need the normalise function

**Comment by lnielsen** [comment]
@jirikuncar We can, but it's a new feature so should probably just go to master?

File: invenio/utils/persistentid.py (hunk 2 above)
Comment (bouzlibop) [inline invenio/utils/persistentid.py]: Isn't that already done in https://github.com/inveniosoftware/invenio/pull/3254/files#diff-f3e79904dc66a9f6d0ebca1960b25154L336?

**Comment by jirikuncar** [comment]
@blixhavn do you need it in `maint-2.0`?

**Comment by lnielsen** [comment]
Closing request as it was integrated in inveniosoftware/idutils

**Pull request closed without merging**